tests/FSharp.Compiler.ComponentTests/Miscellaneous/MigratedCoreTests.fs

```diff
@@ -1,465 +1,477 @@
+    [<Fact>]
+    let ``test int32-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/int32" FSC_OPTIMIZED
 
-[<FactForNETCOREAPP>]
-let ``control-FSI`` () = singleTestBuildAndRun "core/control" FSI
+    [<Fact>]
+    let ``test int32-FSI`` () = singleTestBuildAndRun "core/int32" FSI
 
-[<FactForNETCOREAPP>]
-let ``control --tailcalls`` () =
-    let cfg =  "core/control"
-    singleTestBuildAndRunAux cfg  ["--tailcalls"] FSC_OPTIMIZED
+module Tests6 =
 
-[<Fact>]
-let ``controlChamenos-FSC_OPTIMIZED`` () =
-    let cfg =  "core/controlChamenos"
-    singleTestBuildAndRunAux cfg  ["--tailcalls"] FSC_OPTIMIZED
+    [<Fact>]
+    let ``recordResolution-FSC_DEBUG`` () = singleTestBuildAndRun "core/recordResolution" FSC_DEBUG
 
-[<Fact>]
-let ``controlChamenos-FSI`` () =
-    let cfg =  "core/controlChamenos"
-    singleTestBuildAndRunAux cfg  ["--tailcalls"] FSI
+    [<Fact>]
+    let ``recordResolution-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/recordResolution" FSC_OPTIMIZED
 
-[<Fact>]
-let ``controlMailbox-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/controlMailbox" FSC_OPTIMIZED
+    [<Fact>]
+    let ``recordResolution-FSI`` () = singleTestBuildAndRun "core/recordResolution" FSI
 
-[<Fact(Skip="Flaky")>]
-let ``controlMailbox-FSI`` () = singleTestBuildAndRun "core/controlMailbox" FSI
+    // This test has hardcoded expectations about current synchronization context
+    // Will be moved out of FsharpSuite.Tests in a later phase for desktop framework
+    [<FactForNETCOREAPP>]
+    let ``control-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/control" FSC_OPTIMIZED
 
-[<Fact>]
-let ``controlMailbox --tailcalls`` () =
-    let cfg =  "core/controlMailbox"
-    singleTestBuildAndRunAux cfg  ["--tailcalls"] FSC_OPTIMIZED
+    [<FactForNETCOREAPP>]
+    let ``control-FSI`` () = singleTestBuildAndRun "core/control" FSI
 
-[<Fact>]
-let ``csext-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/csext" FSC_OPTIMIZED
+    [<FactForNETCOREAPP>]
+    let ``control --tailcalls`` () =
+        let cfg =  "core/control"
+        singleTestBuildAndRunAux cfg  ["--tailcalls"] FSC_OPTIMIZED
 
-[<Fact>]
-let ``csext-FSI`` () = singleTestBuildAndRun "core/csext" FSI
+    [<Fact>]
+    let ``controlChamenos-FSC_OPTIMIZED`` () =
+        let cfg =  "core/controlChamenos"
+        singleTestBuildAndRunAux cfg  ["--tailcalls"] FSC_OPTIMIZED
 
-[<Fact>]
-let ``enum-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/enum" FSC_OPTIMIZED
+    [<Fact>]
+    let ``controlChamenos-FSI`` () =
+        let cfg =  "core/controlChamenos"
+        singleTestBuildAndRunAux cfg  ["--tailcalls"] FSI
 
-[<Fact>]
-let ``enum-FSI`` () = singleTestBuildAndRun "core/enum" FSI
+    [<Fact>]
+    let ``controlMailbox-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/controlMailbox" FSC_OPTIMIZED
 
-[<Fact>]
-let ``longnames-FSC_OPTIMIZED`` () = singleTestBuildAndRun "core/longnames" FSC_OPTIMIZED
+    [<Fact>]
```

Comment:
FYI this was turned off quite recently, so it's likely not fixed properly and might start fluking again :/

Reply:
yeah. that thing is the horror. I plan to just write a new test, at least for `TryScan`.